axiom/nr_segment.c

```diff
@@ -313,6 +313,26 @@ static void nr_populate_http_spans(nr_span_event_t* span_event,
                                     segment->typed_attributes->external.status);
 }
 
+static void nr_populate_message_spans(nr_span_event_t* span_event,
+                                      const nr_segment_t* segment) {
+  nr_span_event_set_category(span_event, NR_SPAN_MESSAGE);
```

Comment:
Should this happen before the NULL check below it?

Reply:
The category is being set on the span event, so separate from segment logic at that point (and set_category checks the event for null). Spans can have a type even if no attributes are available (see datastore/external for similar behavior).
This is core logic, so the fewer messages the better, but in this case, no log message is needed because if we are setting span category with no attributes, the absence of attributes would show up both in the UI as well as in the verbosedebug daemon logs.